addon/components/connect.js

```diff
@@ -5,36 +5,46 @@ const {
   computed,
   defineProperty,
   inject: { service },
-  isEmpty,
   run
 } = Ember;
 
 export default (stateToComputed, dispatchToActions=() => ({})) => {
-
-  if (!stateToComputed) {
-    stateToComputed = () => ({});
-  }
-
   return Component => {
-
     return Component.extend({
-
       redux: service(),
 
       init() {
         const redux = this.get('redux');
 
-        let props = stateToComputed.call(this, redux.getState(), this.getAttrs());
+        if (stateToComputed) {
+          const getProps = () => stateToComputed.call(this, redux.getState(), this.getAttrs());
+          let props = getProps();
+
+          Object.keys(props).forEach(name => {
+            defineProperty(this, name, computed(() =>
+              props[name]
+            ).property().readOnly());
+          });
+
+          this._handleChange = () => {
+            const newProps = getProps();
+            if (props === newProps) return;
```

Comment:
Yeah, so this is only useful with things like reselect. Since it memoizes the output of the stateToComputed method, if the state inputs are the same, it'll return the same props object, which means we don't have to do any processing on it!